File: 3229-minimum-operations-to-make-array-equal-to-target/3229-minimum-operations-to-make-array-equal-to-target.py
class Solution:
  def minimumOperations(self, nums: List[int], target: List[int]) -> int:
    a = abs(nums[0] - target[0])
    for (n1, t1), (n2, t2) in itertools.pairwise(zip(nums, target)):
      d1 = t2 - n2
      d2 = t1 - n1
      if d1 >= 0 and d2 >= 0:
        a += max(0, d1 - d2)
      elif d1 <= 0 and d2 <= 0:
        a += max(0, abs(d1) - abs(d2))
      else:
        a += abs(d1)
    return a

File: 3229-minimum-operations-to-make-array-equal-to-target/3229-minimum-operations-to-make-array-equal-to-target.java
class Solution {
  public long minimumOperations(int[] nums, int[] target) {
    long ans = Math.abs((long) nums[0] - target[0]);
    for (int i = 1; i < nums.length; ++i) {
      final int currDiff = target[i] - nums[i];
      final int prevDiff = target[i - 1] - nums[i - 1];
      if (currDiff >= 0 && prevDiff >= 0)
        ans += Math.max(0, currDiff - prevDiff);
      else if (currDiff <= 0 && prevDiff <= 0)
        ans += Math.max(0, Math.abs((long) currDiff) - Math.abs((long) prevDiff));
      else
        ans += Math.abs((long) currDiff);
    }
    return ans;
  }
}